src/domManipulation.ts

```diff
@@ -4,16 +4,27 @@ import fetchWeatherForecast from "./api";
 
 // Get the parent element for the main cities/locations which holds all the buttons.
 const mainCities = document.getElementById("main-cities-nav");
+
 // Get all the button elements in this parent.
-const mainCityButtons = mainCities.getElementsByClassName("city-chip");
+let  mainCityButtons : HTMLCollectionOf<Element>;
+if (mainCities !== null) {
+  mainCityButtons = mainCities.getElementsByClassName("city-chip");
+}
+
 // Set the default city to the first option.
 let selectedMainCity = 0;
 
 // Toggle visibility of the pop-up
 const mainLocations = document.getElementById("main-cities-container");
 const customLocationPopup = document.getElementById("popup-container");
-mainLocations.style.visibility = "visible";
-customLocationPopup.style.visibility = "hidden";
+if (mainLocations !== null) {
+  mainLocations.style.visibility = "visible";
+}
+if (customLocationPopup !== null)
+{
+  customLocationPopup.style.visibility = "hidden";
+}
+
```

Comment:
could be made into a single `if` statement

Reply:
I believe I edited this for the above reviews, I removed this code and edited the closePopup() function as well as the default styling for the popup to hide it when the page loads, which was what this code was doing.